### PR TITLE
Redis sentinel support

### PR DESCRIPTION
**Pull request by Pritesh-Patel**
This is to enable the use of sentinel for redis.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -3,13 +3,14 @@
...
 It also depends on play-json for JSON serialization.
- 
+
 ## Compatibility
 ### Akka 2.4.x and Play 2.5.x
 Use versions from *0.4.0* for Akka 2.4.x and Play 2.5.x
 ```
 resolvers += Resolver.jcenterRepo // Adds Bintray to resolvers for akka-persistence-redis and rediscala
-libraryDependencies ++= Seq("com.hootsuite" %% "akka-persistence-redis" % "0.4.0")
+resolvers += Resolver.bintrayRepo("pritesh-patel", "maven") // to use until merged
```
Comment (ssong-van) [inline README.md]: I can release 0.5.0 immediately after merging the PR

File: src/main/scala/com/hootsuite/akka/persistence/redis/DefaultRedisComponent.scala [hunk 2]
```diff
@@ -3,12 +3,19 @@ package com.hootsuite.akka.persistence.redis
...
-  private val host = config.getString("redis.host")
-  private val port = config.getInt("redis.port")
+  private val sentinel = config.getBoolean("redis.sentinel")
+
+  lazy val redis = if(sentinel){
+    SentinelUtils.getSentinelBasedClient(config)
+  } else {
+    val host = config.getString("redis.host")
+    val port = config.getInt("redis.port")
+    new RedisClient(host, port)
```
Comment (ssong-van) [inline src/main/scala/com/hootsuite/akka/persistence/redis/DefaultRedisComponent.scala]: `new` is not required for a case class

File: src/main/scala/com/hootsuite/akka/persistence/redis/SentinelUtils.scala [hunk 3]
```diff
@@ -0,0 +1,19 @@
...
+
+object SentinelUtils {
+
+  def getSentinelBasedClient(config: Config)(implicit system:ActorSystem):RedisClient = {
+
+    val sentinelMaster = config.getString("redis.sentinel-master")
+    val sentinels = config.getConfigList("redis.sentinels") map { conf =>
+      conf.getString("host") -> conf.getInt("port")
+    }
+    new SentinelMonitoredRedisClient(sentinels,sentinelMaster).redisClient
```
Comment (ssong-van) [inline src/main/scala/com/hootsuite/akka/persistence/redis/SentinelUtils.scala]: Same here. `new` is not required.

File: src/test/scala/com/hootsuite/akka/persistence/redis/journal/RedisJournalSpec.scala [hunk 4]
```diff
@@ -21,9 +22,16 @@ class RedisJournalSpec extends JournalSpec(
   override def beforeAll() = {
     super.beforeAll()
     val config = ConfigFactory.load()
-    val host = config.getString("redis.host")
-    val port = config.getInt("redis.port")
-    val redis = new RedisClient(host, port)
+
+    val sentinel = config.getBoolean("redis.sentinel")
```
Comment (ssong-van) [inline src/test/scala/com/hootsuite/akka/persistence/redis/journal/RedisJournalSpec.scala]: Can you add the sentinel configs to test/resources/reference.conf?

**Comment by ssong-van** [comment]
Thank you so much for your contribution. This looks awesome. 

We ask contributors to sign a CLA and we are rolling out an automated process with https://cla-assistant.io/ I'll work on it today so you can merge it asap. If this is blocking you, I can also send you a Word doc to sign to speed up the process.

**Comment by Pritesh-Patel** [comment]
Could you send me the word doc? :)

**Comment by ssong-van** [comment]
Sure, I opened up google doc for you here - https://docs.google.com/document/d/16AnuPqQIlYCDRvDsWGBGdgmB6QKd-Vo1pu_cqXVzLzE/edit?usp=sharing
Please fill in name, email, and date. Once again, thank you so much!

**Comment by Pritesh-Patel** [comment]
All done!

**Pull request merged**